### DISCUSSION TITLE
[Ideas] Enable pgrx in Cloudberry

### DISCUSSION THREAD

**Discussion by roseduan** [apache/cloudberry#732] [Ideas / Feature Requests] [answered]
### Description

pgrx(https://github.com/pgcentralfoundation/pgrx) is a framework for developing PostgreSQL extensions in Rust.
it supports Postgres 12 through Postgres 17.

As we know, Cloudberry`s pg version is 14.4, so it is possible to enable pgrx.

By doing this, we can build new extensions in Cloudberry with rust, and import other extensions in rust(like iceberg-rust, and other datalake components) to enrich the Cloudberry ecosystem.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by yihong0618** [comment]
Its not quick hard for doing this, only add both big-endian vs little-endian for pgrx side is OK.
And its already done on gpdb side before~

**Comment by tuhaihe** [comment]
For better context, I would like to share 2 links related to this topic for your information:

- https://github.com/pgcentralfoundation/pgrx/issues/1179
- https://github.com/pgcentralfoundation/pgrx/issues/1952

**Comment by roseduan** [comment, marked as answer]
Work in progress: https://github.com/cloudberry-contrib/pgrx
